lib/rubygems/gemcutter_utilities.rb

```diff
@@ -267,6 +270,25 @@ def get_scope_params(scope)
     scope_params
   end
 
+  def get_mfa_params(email, password)
+    mfa_level = get_user_mfa_level(email, password)
+    params = {}
+    if mfa_level == "ui_only" || mfa_level == "ui_and_gem_signin"
+      selected = ask "Would you like to enable MFA for this key? [Y/n]"
+      params["mfa"] = true unless selected =~ /^[nN](o)?$/
+    end
+    params
+  end
+
+  def get_user_mfa_level(email, password)
+    response = rubygems_api_request(:get, "api/v1/profile/me") do |request|
```

Comment:
Any idea what's going to happen when endpoint fails (for example when is not implemented at other than rubygems.org gemservers)?